### PR TITLE
Use proper di in cloud discovery

### PR DESCRIPTION
**Pull request by arbulu89**
Use proper dependency injection with the command executor in the cloud discovery code

### CODE REVIEW

File: internal/cloud/metadata.go [hunk 1]
```diff
@@ -22,16 +22,22 @@ type Instance struct {
...
+	return &Identifier{
+		executor: executor,
+	}
+}
 
 // All these detection methods are based in crmsh code, which has been refined over the years
 // https://github.com/ClusterLabs/crmsh/blob/master/crmsh/utils.py#L2009
 
-func identifyAzure() (bool, error) {
+func (c *Identifier) identifyAzure() (bool, error) {
```
Comment (rtorrero) [inline internal/cloud/metadata.go:38]: What does the `c` stand for?

File: internal/cloud/metadata.go [hunk 2]
```diff
@@ -83,24 +89,24 @@ func identifyGcp() (bool, error) {
...
-	if result, err := identifyAzure(); err != nil {
+	if result, err := c.identifyAzure(); err != nil {
 		return "", err
 	} else if result {
 		log.Infof("VM is running on %s", Azure)
 		return Azure, nil
 	}
 
-	if result, err := identifyAws(); err != nil {
+	if result, err := c.identifyAws(); err != nil {
```
Comment (rtorrero) [inline internal/cloud/metadata.go:102]: This is probably out of scope for this PR so feel free to ignore, but should it be `identifyAWS` instead?

File: internal/cloud/metadata.go [hunk 3]
```diff
@@ -83,24 +89,24 @@ func identifyGcp() (bool, error) {
...
 		return "", err
 	} else if result {
 		log.Infof("VM is running on %s", Aws)
 		return Aws, nil
 	}
 
-	if result, err := identifyGcp(); err != nil {
+	if result, err := c.identifyGcp(); err != nil {
```
Comment (rtorrero) [inline internal/cloud/metadata.go:109]: Same as previous

File: internal/cluster/cluster.go [hunk 4]
```diff
@@ -116,7 +119,11 @@ func NewClusterWithDiscoveryTools(discoveryTools *DiscoveryTools) (Cluster, erro
 
 	cluster.DC = isDC(&cluster)
 
-	provider, _ := cloud.IdentifyCloudProvider()
+	cloudIdentifier := cloud.NewIdentifier(utils.Executor{})
+	provider, err := cloudIdentifier.IdentifyCloudProvider()
+	if err != nil {
+		log.Warn(errors.Wrap(err, "Cloud provider not identified"))
```
Comment (rtorrero) [inline internal/cluster/cluster.go:125]: I'd write: `Couldn't identify cloud provider`

**Review by rtorrero — APPROVED** [review]
Thanks @arbulu89! Giving green light as my comments are mostly cosmetic or just questions.

File: internal/cloud/metadata.go (hunk 1 above)
Comment (arbulu89) [reply internal/cloud/metadata.go:38]: `Cloud`. I will change it by `i`

**Pull request merged**